### DISCUSSION TITLE
Developing a simple algorithm

### DISCUSSION THREAD

**Discussion by lpmrz** [ctu-mrs/mrs_uav_system#104] [Q&A] [answered]
Hello,

I want to implement a formation control algorithm that I have written in MATLAB. I am trying to understand the workflow of the MRS system. 

In the documentation part, I have seen how to spawn and command the drones (takeoff, land, start following a trajectory) by calling services from the terminal. But I couldn't find a page (maybe I missed it) for writing a C++ or Python code. Right now I just want to write a code(a rosservice) that moves the drone to a specific position. Later I will utilize sensory information and develop a proper algorithm.

Thank you in advance. 

edit#1: For starting, I can try to implement in Python, maybe there is a list of functions I call and libraries I should import.

edit#2: I solved my problem for now by writing my code in Python. I am using rospy.ServiceProxy().

**Comment by klaxalk** [comment, marked as answer]
Hello, we encourage our users to explore the possibilities by themselves. Please, learn to search for answers before you pose a question. I recommend searching for examples (not limited to our repositories). I hope you were able to solve that already. Sincerely, Tomas

**Reply by lpmrz** [reply]
Hello, yes I was able to solve it before. Thank you for your reply.
